libnmstate/ifaces/ifaces.py

```diff
@@ -180,6 +181,24 @@ def _validate_ovs_patch_peers(self):
                             " patch port"
                         )
 
+    def _validate_vlan_not_over_infiniband(self):
+        """
+        Validate that vlan is not being created over infiniband interface
+        """
+        for iface in self._ifaces.values():
+
+            if (
+                iface.type in [InterfaceType.VLAN, InterfaceType.VXLAN]
+                and iface.is_up
+            ):
+                base_iface = self._ifaces.get(iface.parent)
```

Comment:
Previous validation functions has confirmed the existence of parent interface. Just use `self._ifaces[iface.parent]`.